presto-docs/src/main/sphinx/security/server.rst

```diff
@@ -0,0 +1,254 @@
+  > ktadd -k /etc/presto/presto.keytab presto/presto-coordinator.example.com@EXAMPLE.COM
+
+.. include:: ktadd-note.fragment
+
+.. include:: jce-policy.fragment
+
+.. _coordinator_java_keystore:
+
+Java Keystore File for TLS
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Access to the Presto coordinator should be through https when using Kerberos
+authentication. The Presto coordinator needs keys to secure the :abbr:`TLS
+(Transport Layer Security)` connection. These keys are generated using
+:command:`keytool` and stored in a Java Keystore file for the Presto
+coordinator.
+
+.. code-block:: none
+  
+  keytool -genkeypair -alias presto -keyalg RSA -keystore keystore.jks
+  Enter keystore password:
+  Re-enter new password:
+  What is your first and last name?
+    [Unknown]:  presto-coordinator
+  What is the name of your organizational unit?
+    [Unknown]:
+  What is the name of your organization?
+    [Unknown]:
+  What is the name of your City or Locality?
+    [Unknown]:
+  What is the name of your State or Province?
+    [Unknown]:
+  What is the two-letter country code for this unit?
+    [Unknown]:
+  Is CN=eiger, OU=Unknown, O=Unknown, L=Unknown, ST=Unknown, C=Unknown correct?
+    [no]:  yes
+
+  Enter key password for <presto>
+          (RETURN if same as keystore password):  
+
+The alias should match the principal that the presto server will use.
+
+The prompt for first and last name is actually prompting for the Common Name
+that will be used in the certificate. In this case, it should be the
+unqualified hostname of the Presto coordinator. You can see this in the prompt
+that confirms that the information is crorrect.
+
+Access Controller Implementation
+--------------------------------
+
+Presto separates the concept of the principal who authenticates to the
+coordinator from the username that is responsible for running queries. When
+running the Presto CLI, for example, the Presto username can be specified using
+the --user option.
+
+By default, the Presto coordinator allows any principal to run queries as any
+Presto user. In a secure environment, this is probably not desirable behavior.
+
+This behavior can be customized by implementing the
+``SystemAccessControlFactory`` and ``SystemAccessControl`` interfaces.
+
+``SystemAccessControlFactory`` is responsible for creating a
+``SystemAccessControl`` instance. It also defines a ``SystemAccessControl``
+name which is used by the administrator in a Presto configuration.
+
+``SystemAccessControl`` implementations have two responsibilities:
+ * Verifying whether or not a given Kerberos principal is authorized to execute
+   queries as a specific user.
+ * Determining whether or not a given user can alter values for a given system
+   property.
+
+The implementation of ``SystemAccessControl`` and
+``SystemAccessControlFactory`` must be wrapped as a plugin and installed on the
+Presto cluster.
+
+Presto Coordinator Node Configuration
+-------------------------------------
```

Comment:
Do all the other changes need to be made before you configure the Presto coordinator node?